tools/pipelines/test-real-service.yml

```diff
@@ -128,6 +140,68 @@ stages:
         FLUID_TEST_LOGGER_PKG_SPECIFIER: '@ff-internal/aria-logger' # Contains createTestLogger impl to inject
         FLUID_LOGGER_PROPS: '{ "displayName": "${{variables.pipelineIdentifierForTelemetry}}"}'
 
+  # end-to-end tests docker
+  - template: templates/include-test-real-service.yml
+    parameters:
+      stageId: e2e_docker
+      stageDisplayName: e2e - docker
+      poolBuild: Large
+      testPackage: ${{ variables.testPackage }}
+      testWorkspace: ${{ variables.testWorkspace }}
+      artifactBuildId: $(resources.pipeline.client.runID)
+      continueOnError: true
+      testCommand: test:realsvc:r11s:docker
+      checkout: true
+      stageVariables:
+        - group: container-registry-info
+      splitTestVariants:
+        - name: Non-compat
+          flags: --compatVersion=0
+        - name: N-1
+          flags: --compatVersion=-1
+        - name: LTS
+          flags: --compatVersion=LTS
+        - name: Cross-Version
+          flags: --compatVersion=CROSS_VERSION
+      cacheCompatVersionsInstalls: true
+      uploadTestPassRateTelemetry: true
+      pipelineIdentifierForTelemetry: ${{ variables.pipelineIdentifierForTelemetry }}
+      preSteps:
+        - task: Docker@2
+          displayName: 'Login to container registry'
+          inputs:
+            # $(containerRegistryConnection) comes from the container-registry-info variable group and needs to be
+            # specified as a runtime variable (variables from variable groups apparently are never available "statically"
+            # at parse/compile time, so can't be used with template-expression syntax ( '${{ }}' )).
+            containerRegistry: 'Fluid Container Registry' # $(containerRegistryConnection)
+            command: 'login'
+        - task: DockerCompose@1
+          displayName: 'Start routerlicious environment in docker'
+          inputs:
+            containerregistrytype: 'Container Registry'
+            dockerComposeFile: 'server/docker-compose.yml'
+            action: 'Run a Docker Compose command'
+            dockerComposeCommand: 'up -d'
+            projectName: 'routerlicious'
+          env:
+            REGISTRY_URL: fluidcr.azurecr.io/build
+            ALFRED_IMAGE_TAG: 0.0.$(resources.pipeline.routerlicious.runID)
+            HISTORIAN_IMAGE_TAG: 0.0.$(resources.pipeline.historian.runID)
+            GITREST_IMAGE_TAG: 0.0.$(resources.pipeline.gitrest.runID)
+            GITSSH_IMAGE_TAG: 0.0.$(resources.pipeline.gitssh.runID)
+      additionalSteps:
+        - task: DockerCompose@1
+          displayName: 'Stop routerlicious environment in docker'
+          condition: always()
+          inputs:
+            containerregistrytype: 'Container Registry'
+            dockerComposeFile: 'server/docker-compose.yml'
+            action: 'Run a Docker Compose command'
+            dockerComposeCommand: 'down'
+            projectName: 'routerlicious'
+      env:
```

Comment:
Something new we put in `env` in all stages after I opened the original PR is `FLUID_LOGGER_PROPS: '{ "displayName": "${{variables.pipelineIdentifierForTelemetry}}"}'`, we should add it in this new one too.